fix(curriculum): separate unreachable code and scope for pyramid project (#54920)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c71600966a2191d3a64a.md

# --description--

> Returning a value from a function brings that value into the scope where the function was called. To bring your `"Testing"` value from the `padRow` function into the global scope, update your return statement to return only the `test` variable.

Reviewer: ````md
Values returned out of functions are used by calling the function. You can use the function call directly as the value it returns, or capture the returned value in a variable. This way, you can use the value assigned to a locally scoped variable, outside the function it was created in.

```js
function getName() {
  const name = "Camper cat";
  return name;
}

console.log(getName()); // "Camper cat"

const capturedReturnValue = getName();
console.log(capturedReturnValue); // "Camper cat"

console.log(name); // reference error
```

To use your `"Testing"` value returned from the `padRow` function, update your `return` statement to return only the `test` variable.

````

---

This is just a suggestion, but I think what we have now can be misinterpreted in multiple ways. I suggest we simplify this and give a small code example.

# --description--

Values returned out of a function are used by calling the function. You can use the function call directly as the value it returns, or capture the returned value in a variable. This way, you can use the value assigned to a locally scoped variable, outside the function it was created in.

```js
function getName() {
  const name = "Camper cat";
  return name;
}

console.log(getName()); // "Camper cat"

const capturedReturnValue = getName();
console.log(capturedReturnValue); // "Camper cat"

console.log(name); // reference error
```

> To use your `"Testing"` value returned from the `padRow` function, update your `return` statement to return only the `test` variable.

Reviewer: ```suggestion
To use your `"Testing"` value, return it out of the `padRow` function by updating your `return` statement to return only the `test` variable.
```

Sorry, I missed this when I updated the text.
